server/src/trainee/traineeController.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Definition of the handler functions for the members module
+ */
+import bcrypt = require("bcryptjs");
+import { Request, Response } from "express";
+import database = require("../database");
+import { getRandomString } from "../utils/stringUtils";
+import { createMailAccount, addMailAccountToMailingList } from "../utils/plesk";
+import { createMWUser } from "../utils/mediawiki";
+import { createNCUser } from "../utils/nextcloud";
+import * as traineeTypes from "./traineeTypes";
+import { PoolConnection } from "mysql";
+import * as authTypes from "./../global/auth/authTypes";
+import { canPermissionBeDelegated, doesPermissionsInclude } from "../utils/authUtils";
+
+/**
+ * Retrieves choices of mentor, internal project and department of all trainees of given generation
+ * @param req generation_ID
+ * @param res member ID, first and last name and choices
+ */
+export const retrieveTraineeChoice = (req: Request, res: Response): void => {
+  database
+    .query(
+      `SELECT mitgliedID, vorname, nachname,
+      wahl_mentor, wahl_mentor1, wahl_mentor2, wahl_mentor3,
+      wahl_internesprojekt, wahl_internesprojekt1, wahl_internesprojekt2, wahl_internesprojekt3,
+      wahl_ressort, wahl_ressort1, wahl_ressort2, wahl_ressort3
+      FROM  mitglied
+      INNER JOIN generation
+      ON mitglied.generation = generation.generationID
+      WHERE  generation.generationID = ?`,
+      [req.body.generation_ID]
+    )
+    .then((result: traineeTypes.GetTraineeChoiceResult) => {
+      res.status(200).json(result);
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Gets letter of motivation form trainees of given generation
+ * @param req generation_ID
+ * @param res member ID and 3 motivation texts
+ */
+export const retrieveTraineeMotivation = (req: Request, res: Response): void => {
+  database
+    .query(
+      `SELECT mitgliedID, wahl_internesprojekt1_motivation, wahl_internesprojekt2_motivation,
+      wahl_internesprojekt3_motivation
+      FROM mitglied
+      INNER JOIN generation
+      ON mitglied.generation = generation.generationID
+      WHERE generation.generationID = ?`,
+      [req.body.generation_ID]
```

Comment:
Please use camel case variables:
```suggestion
      [req.body.generationID]
```

Reply:
fixed